### PR TITLE
Fix dims docstring for torch.flip and torch.rot90 (axes, element type)

### LINKED ISSUE #130641: `dims` of `flip()` without `list` or `tuple` works with a tensor agaist the doc
### 📚 The doc issue

[The doc](https://pytorch.org/docs/stable/generated/torch.flip.html) of `flip()` says the type of `dims` is `list` or `tuple` as shown below:

> - dims (a list or [tuple](https://docs.python.org/3/library/stdtypes.html#tuple)) – axis to flip on

But, `dims` without `list` or `tuple` works with a tensor against the doc as shown below:

```python
import torch

my_tensor = torch.tensor([[[2, 7, 4], [8, 3, 2]], # 3D tensor
                          [[5, 0, 8], [3, 6, 1]]])
my_tensor.flip(0, 1, 2)
# tensor([[[1, 6, 3],
#          [8, 0, 5]],
#         [[2, 3, 8],
#          [4, 7, 2]]])
```

### Suggest a potential alternative/fix

_No response_

### PR DESCRIPTION
**Pull request by vineethsaivs**
Fixes #130641.

`torch.flip`'s docstring describes `dims` as:

```
dims (a list or tuple): axis to flip on
```

`dims` is a list/tuple of dimensions, so "axis" should be plural ("axes"), and the element type can be named, consistent with how `torch.permute` documents its `dims` argument (`tuple of int or list of int`). `torch.rot90` in the same file has the identical wording (`axis to rotate`), so this fixes both:

```
- dims (a list or tuple): axis to flip on
+ dims (a list or tuple of ints): axes to flip on

- dims (a list or tuple): axis to rotate. Default value is [0, 1]
+ dims (a list or tuple of ints): axes to rotate. Default value is [0, 1]
```

Docstring-only, no runtime change. (Like `torch.permute`'s functional docstring, this documents the list/tuple form; the positional-varargs convenience for single `int[]` arguments is a general behavior not spelled out at the functional level.)

### CODE REVIEW

**Review by jbschlosser — APPROVED** [review]

**Comment by vineethsaivs** [comment]
Thanks for the approve + merge @jbschlosser! The merge failed only on unrelated GPU jobs: `linux-jammy-rocm-py3.10-mi350` (the ROCm MI350 SEV) and `macos-py3-arm64 (mps)`. This PR is a docstring-only change to `torch/_torch_docs.py` (+2/-2, no runtime/code change), so those can't be affected by it. Would you mind re-running with `@pytorchbot merge -i` to ignore the current unrelated failures?

**Comment by vineethsaivs** [comment]
Gentle ping @jbschlosser: the only failures are still the `linux-jammy-rocm-py3.10-mi350` shards from the ongoing ROCm SEV (175 other jobs pass, and this is a +2/-2 docstring-only change). `@pytorchbot merge -i` when you get a chance would land it. Thanks!

**Comment by vineethsaivs** [comment]
Rebased onto `viable/strict` so CI re-runs clean, the only failures on the previous head were the stale `linux-jammy-rocm-py3.10-mi350` shards from the June ROCm SEV plus `macos-py3-arm64 (mps)`, both unrelated to a +2/-2 docstring change in `torch/_torch_docs.py`.

@jbschlosser could you re-run `@pytorchbot merge` once the new run settles? Your approval from 29 Jun is still on it.
